modules/apps/client-extension/client-extension-web/src/main/resources/META-INF/resources/js/components/AttributeFields.tsx

```diff
@@ -0,0 +1,186 @@
+/**
+ * SPDX-FileCopyrightText: (c) 2000 Liferay, Inc. https://liferay.com
+ * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
+ */
+
+import {ClayButtonWithIcon} from '@clayui/button';
+import {Option, Picker} from '@clayui/core';
+import {ClayInput} from '@clayui/form';
+import ClayLayout from '@clayui/layout';
+import classNames from 'classnames';
+import {FieldBase} from 'frontend-js-components-web';
+import React, {useState} from 'react';
+
+interface AttributeFieldsProps {
+	index: number;
+	name: string;
+	onAddClick: (index: number) => void;
+	onAttributeChange: (index: number, updatedValue: Object) => void;
+	onRemoveClick: (index: number) => void;
+	portletNamespace: string;
+	type: string;
+	value: string | boolean;
+}
+
+const BOOLEAN_VALUE_ITEMS = [
+	{label: Liferay.Language.get('true'), value: true},
+	{label: Liferay.Language.get('false'), value: false},
+];
+
+const TYPE_ITEMS = [
+	{label: Liferay.Language.get('string'), value: 'String'},
+	{label: Liferay.Language.get('boolean'), value: 'Boolean'},
+];
+
+const AttributeFields: React.FC<AttributeFieldsProps> = ({
+	index,
+	name,
+	onAddClick,
+	onAttributeChange,
+	onRemoveClick,
+	portletNamespace,
+	type,
+	value,
+}) => {
+	const [errorMessage, setErrorMessage] = useState<string | null>(null);
+
+	return (
+		<ClayLayout.Row className="mb-3">
+			<ClayLayout.Col
+				className={classNames({'has-error': errorMessage})}
+				size={4}
+			>
+				<FieldBase
+					className="mb-0"
+					errorMessage={errorMessage}
+					id={`${portletNamespace}name_${index}`}
```

Comment:
@ilzamcmed Please create a variable for strings like this that are used more than once so that we don't have to worry about typos between them.